entity-view/impl/src/main/java/com/blazebit/persistence/view/impl/mapper/ViewMapper.java

```diff
@@ -381,12 +391,13 @@ public Key(ManagedViewTypeImplementor<S> sourceType, ManagedViewTypeImplementor<
         }
 
         public static <Y> Key<Object, Y> create(ViewMetamodel metamodel, Object source, Class<Y> targetEntityViewClass, String constructorName, ConvertOption... convertOptions) {
-            if (!(source instanceof EntityViewProxy)) {
-                throw new IllegalArgumentException("Can only convert one entity view to another. Invalid source type: " + source.getClass());
+            if (source instanceof EntityViewProxy) {
+                EntityViewProxy sourceProxy = (EntityViewProxy) source;
+                return (Key<Object, Y>) create(metamodel, sourceProxy.$$_getEntityViewClass(), targetEntityViewClass, constructorName, convertOptions);
+            } else {
+                return (Key<Object, Y>) create(metamodel, null, targetEntityViewClass, constructorName, convertOptions);
```

Comment:
IMO we should validate that the entity class of the target entity view is a super type of the source object.

```suggestion
            ManagedViewType<?> targetViewType = metamodel.managedView(targetEntityViewClass);
            if (targetViewType == null) {
                throw new IllegalArgumentException("Unknown target view type: " + targetEntityViewClass.getName());
            }
            if (!targetViewType.getEntityClass().isInstance(source)) {
                throw new IllegalArgumentException("The source object is not an instance of the target views entity type " + targetViewType.getEntityClass().getName() + ": " + source.getClass().getName());
            }
            MappingConstructorImpl<Y> constructor = null;
            if (targetConstructorName != null) {
                constructor = (MappingConstructorImpl<Y>) targetViewType.getConstructor(targetConstructorName);
                if (constructor == null) {
                    throw new IllegalArgumentException("Unknown constructor '" + targetConstructorName + "' on target view type: " + targetEntityViewClass.getName());
                }
            }
            boolean ignoreMissingAttributes = false;
            boolean markNew = false;
            for (ConvertOption copyOption : convertOptions) {
                switch (copyOption) {
                    case CREATE_NEW:
                        markNew = true;
                        break;
                    case IGNORE_MISSING_ATTRIBUTES:
                        ignoreMissingAttributes = true;
                        break;
                    default:
                        break;
                }
            }
            return new ViewMapper.Key<>(sourceViewType, targetViewType, constructor, ignoreMissingAttributes, markNew);
```